### PR TITLE
Add EL8 support

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Review by wbclark — COMMENTED** [review]
acceptance tests didn't run here because they aren't added until https://github.com/theforeman/puppet-qpid/pull/154

therefore I did a checkout of that branch and cherry-picked this commit on top of it. running the tests locally on EL8, it fails with:

`  Error: Execution of '/usr/bin/dnf -d 0 -e 1 -y install qpid-cpp-server' returned 1: Error: Unable to find a match`

the reason for that seems to be that we only install EPEL on EL7: https://github.com/theforeman/puppet-qpid/pull/154#diff-bc6ad36e1eaefacb5face10b0d75de01df4c7bc7611f63d907ae5c8a2633f7acR9-R12

Even when I do enable EPEL on my test system, this still fails with

```
# dnf install qpid-cpp-server
Repository 'powertools' is missing name in configuration, using id.
Cannot find a valid baseurl for repo: powertools, ignoring this repo.
Last metadata expiration check: 0:00:39 ago on Mon 05 Apr 2021 04:37:21 PM UTC.
No match for argument: qpid-cpp-server
Error: Unable to find a match
```

the available packages matching 'qpid' were:

```
[root@centos8-64-1 ~]# dnf search qpid
Repository 'powertools' is missing name in configuration, using id.
Cannot find a valid baseurl for repo: powertools, ignoring this repo.
Last metadata expiration check: 0:02:27 ago on Mon 05 Apr 2021 04:37:21 PM UTC.
========================================================================== Name & Summary Matched: qpid ==========================================================================
qpid-proton-tests.noarch : Qpid Proton Tests
qpid-proton-c.x86_64 : C libraries for Qpid Proton
qpid-proton-cpp.x86_64 : C++ libraries for Qpid Proton
qpid-dispatch-tools.noarch : Tools for the Qpid Dispatch router
qpid-dispatch-router.x86_64 : The Qpid Dispatch Router executable
qpid-dispatch-console.noarch : Web console for Qpid Dispatch Router
qpid-dispatch-docs.noarch : Documentation for the Qpid Dispatch router
qpid-proton-c-docs.noarch : Documentation for the C development libraries for Qpid Proton
centos-release-qpid-proton.noarch : Config to enable the Messaging SIG QPID Proton packages
rubygem-qpid_proton.x86_64 : Ruby language bindings for the Qpid Proton messaging framework
python3-qpid-proton.x86_64 : Python language bindings for the Qpid Proton messaging framework
qpid-proton-cpp-docs.noarch : Documentation for the C++ development libraries for Qpid Proton
qpid-proton-c-devel.x86_64 : Development libraries for writing messaging apps with Qpid Proton
python-qpid-proton-docs.noarch : Documentation for the Python language bindings for Qpid Proton
qpid-proton-cpp-devel.x86_64 : Development libraries for writing messaging apps with Qpid Proton
[root@centos8-64-1 ~]# 
```

**Comment by wbclark** [comment]
With https://github.com/theforeman/puppet-qpid/commit/22409221bbd6e3f490253e315fa5e1ab6b34a9d2 merged, we should now be able to see acceptance tests run on EL8 with this PR

**Comment by ehelms** [comment]
I rebased and updated the acceptance tests for EL8.

File: spec/setup_acceptance_node.pp [hunk 1]
```diff
@@ -6,6 +6,11 @@
   yumrepo { 'powertools':
     enabled => true,
   }
+
+  yumrepo { 'katello':
+    baseurl  => "http://yum.theforeman.org/katello/nightly/katello/el8/x86_64/",
+    gpgcheck => 0,
+  }
```
Comment (ehelms) [inline spec/setup_acceptance_node.pp:13]: On EL8, we build into and use packages from Katello's repository.

**Review by wbclark — APPROVED** [review]
Looks good. Thanks for the explanation about the change in how we handle packaging for EL8

**Pull request merged**